components/style/context.rs

```diff
@@ -38,6 +38,10 @@ pub struct SharedStyleContext {
     /// Screen sized changed?
     pub screen_size_changed: bool,
 
+    /// Skip the root during traversal?
+    #[cfg(feature = "gecko")]
+    pub skip_root: bool,
```

Comment:
This needs a pretty big comment about why is it needed, if it's going to stay around.